### PR TITLE
Count commits when no tags are present (#72)

### PR DESCRIPTION
**Pull request by raboof**
#72

### CODE REVIEW

**Comment by dwijnand** [comment]
Scala 2.10 troubles 😕:

```scala
[error] /home/travis/build/[secure]/sbt-dynver/src/main/scala/sbtdynver/DynVerPlugin.scala:161:12: type mismatch;
[error]  found   : x$9.type (with underlying type String)
[error]  required: ?{def lines: ?}
[error] Note that implicit conversions are not applicable because they are ambiguous:
[error]  both method augmentString in object Predef of type (x: String)scala.collection.immutable.StringOps
[error]  and method stringToProcess in trait ProcessExtra of type (command: String)sbt.ProcessBuilder
[error]  are possible conversion functions from x$9.type to ?{def lines: ?}
[error]       .map(_.lines.size)
[error]            ^
[error] /home/travis/build/[secure]/sbt-dynver/src/main/scala/sbtdynver/DynVerPlugin.scala:161:14: value lines is not a member of String
[error]       .map(_.lines.size)
[error]              ^
```

**Comment by dwijnand** [comment]
Thanks @raboof!

File: src/main/scala/sbtdynver/DynVerPlugin.scala [hunk 1]
```diff
@@ -154,11 +154,23 @@ sealed case class DynVer(wd: Option[File]) {
   def isDirty(): Boolean                  = getGitDescribeOutput(new Date).isDirty
   def hasNoTags(): Boolean                = getGitDescribeOutput(new Date).hasNoTags
 
+  def getDistanceToFirstCommit() = {
+    val process = scala.sys.process.Process(s"git log --pretty=oneline --abbrev-commit", wd)
+    Try(process !! impl.NoProcessLogger).toOption
```
Comment (agemooij) [inline src/main/scala/sbtdynver/DynVerPlugin.scala]: @raboof Much faster: `git rev-list --count HEAD`

Comment (dwijnand) [reply src/main/scala/sbtdynver/DynVerPlugin.scala]: #git-perf-opt-tips

Comment (raboof) [reply src/main/scala/sbtdynver/DynVerPlugin.scala]: cool I'll do an update!

Comment (raboof) [reply src/main/scala/sbtdynver/DynVerPlugin.scala]: #74

**Pull request merged**